### PR TITLE
[expo-app-auth] Add bare-workflow/standalone support

### LINKED ISSUE #5488: AppAuth on Android breaks when switching app during AppAuth.authAsync
### Environment

```  Expo CLI 3.0.8 environment info:
    System:
      OS: macOS 10.14.6
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 10.14.1 - ~/.nvm/versions/node/v10.14.1/bin/node
      Yarn: 1.15.2 - /usr/local/bin/yarn
      npm: 6.4.1 - ~/.nvm/versions/node/v10.14.1/bin/npm
    IDEs:
      Xcode: 10.3/10G8 - /usr/bin/xcodebuild
    npmPackages:
      @types/react: ^16.8.23 => 16.8.23 
      @types/react-native: ^0.57.65 => 0.57.65 
      @types/react-navigation: ^3.0.7 => 3.0.7 
      expo: 34.0.4 => 34.0.4 
      react: 16.8.6 => 16.8.6 
      react-native: https://github.com/expo/react-native/archive/sdk-34.0.0.tar.gz => 0.59.8 
      react-navigation: ^3.11.1 => 3.11.1 
    npmGlobalPackages:
      expo-cli: 3.0.8
```
Affects:
- Android (Expo Client)
- Android (Simulator, Standalone)
- Galaxy S8 (Standalone)

Versions:
expo-app-auth: 6.0.0 (Snack is 5.0.0)

### Steps to Reproduce

In an app that uses AppAuth to authenticate the user (we are using Auth0):

1. Open app
2. Click the Log In button that triggers `AppAuth.authAsync()`
3. The web browser opens
4. Press the Android Home button
5. Re-open the app from the icon on the home screen (Standalone) or the Snack in the Expo client
6. The app is now back on the screen with the Log In button
7. The Log In button stopped working and triggers the exception `Error: ExpoAppAuth.Get Auth: Cannot start a new task while another task is currently in progress` and user cannot return to the browser window.

### Expected Behavior

Two possible expected behaviors:

1. The user is returned to the active browser sessions when clicking the app-icon on Android
2. When opening the app again the AppAuth session is canceled and the user can use the button to start a new Auth session.

### Actual Behavior

Without any user feedback present, the user is presented with a non-functioning button.

### Reproducible Demo

This snack reproduces the issue: https://snack.expo.io/@platformco/appauth-android-bug

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- Possibly fixes: #5488
- Adds bare-workflow support

# How

- Added a redirect that doesn't require the kernel.

# Test Plan

Important to do, not sure of a continuous way to do this in CI tho.

- [x] test Android in NCL.

### CODE REVIEW

File: packages/expo-app-auth/android/src/main/java/expo/modules/appauth/AppAuthModule.java [hunk 1]
```diff
@@ -270,52 +292,53 @@ private void authWithConfiguration(
...
+      mAuthTask.reject(error);
+      return;
+    }
 
-      if (scopes != null) {
-        String scopesString = Serialization.scopesToString(scopes);
-        if (scopesString != null) {
-          authReqBuilder.setScope(scopesString);
-        }
+    AuthorizationRequest.Builder authReqBuilder = new AuthorizationRequest.Builder(serviceConfig, clientId, ResponseTypeValues.CODE, Uri.parse(redirectUrl));
```
Comment (esamelson) [inline packages/expo-app-auth/android/src/main/java/expo/modules/appauth/AppAuthModule.java]: looks like you removed the `responseType` parsing and the try-catch block i added in https://github.com/expo/expo/pull/6716 -- was that intentional?

**Review by esamelson — COMMENTED** [review]
one question but otherwise lgtm. eventually would be nice to not have to special case the Expo client so much, but I think this is fine for now.

File: packages/expo-app-auth/android/src/main/java/expo/modules/appauth/AppAuthModule.java (hunk 1 above)
Comment (esamelson) [reply packages/expo-app-auth/android/src/main/java/expo/modules/appauth/AppAuthModule.java]: oh, I see now that this PR is pretty old, which explains why that stuff is missing. If you could rebase/merge with master so this PR doesn't unintentionally revert #6716, that would be great!

**Review by esamelson — APPROVED** [review]

**Comment by EvanBacon** [comment]
Closing in favor of #7450. AFAICT the `expo-web-browser` implementation of managed/bare redirects work more reliably.

**Pull request closed without merging**